protected/commands/FilesCommand.php

```diff
@@ -0,0 +1,68 @@
+<?php
+/**
+ * Command to check fils url
+ *
+ * @author Rija Menage <rija+git@cinecinetique.com>
+ * @license GPL-3.0
+ */
+
+use \yii\console\ExitCode;
+use \Codeception\Util\HttpCode;
+
+class FilesCommand extends CConsoleCommand
+{
+    const RETURN_ASSOCIATIVE_ARRAY = 1 ;
+
+    /**
+     * @return string
+     */
+    public function getHelp()
+    {
+        $helpText = "checks files url for a specific dataset in the database" . PHP_EOL;
+        $helpText .= "Usage: ./protected/yiic files checkUrls --doi=<DOI>" . PHP_EOL;
+
+        return $helpText;
+    }
+
+    /**
+     *
+     * Query databse for files url associated to dataset passed as parameter and check that they resolve ok
+     * otherwise output the url
+     *
+     * @param $doi string identifier of the dataset for which to check url
+     * @return int
+     * @throws CException
+     */
+    public function actionCheckUrls($doi) {
+        $sql =<<<END
+SELECT f.location
+FROM file f, dataset d
+WHERE f.dataset_id = d.id and d.identifier = '$doi';
+END;
+
+        try {
+            $rows = Yii::app()->db->createCommand($sql)->queryAll();
+            foreach ($rows as $row) {
+                $parts = parse_url($row['location']);
+                if ( "ftp" === $parts['scheme']) {
+                    echo $row['location'].PHP_EOL;
+                    continue;
+                }
+                $headers = get_headers($row['location'], self::RETURN_ASSOCIATIVE_ARRAY, stream_context_create(array('http' => array('method' => 'HEAD'))));
+                if ($headers[0] !== HttpCode::OK)
```

Comment:
The expected behaviour is that if a file URL resolves OK then it will not be outputted. However, resolvable file URLs are printed in my terminal when I run the tool, e.g. `https://ftp.cngb.org/pub/gigadb/pub/10.5524/100001_101000/100020/readme.txt` when I run this command against the `dev` GigaDB website:
```
$ docker-compose exec application  ./protected/yiic files checkUrls --doi=100020
```

Maybe the problem is because the value of `HttpCode::OK` is 200 which is not equal to the `HTTP/1.1 200 OK` value of `$headers[0]` for a resolvable URL. The `!==` comparison operator will therefore always output `true` even if the file URL is resolvable.

Reply:
Thanks @pli888, your observation was correct and  the last commit should have fixed the problem.